File: responses/QFB7HE-A7D/longest_common_prefix.rb
# you can write to stdout for debugging purposes, e.g.
# puts "this is a debug message"

def solution(a)
  str = ''
  a.each do |item|
    i = 0
    while i < item.length
      matched = 0

      a.each do |string|
        matched += 1 if string.include?(str)
      end

      if matched == a.length && i > 0
        str += item[i-1]
      else
        break
      end
      i += 1
    end
  end
  return str
end

File: responses/M7BJS4-GTC/longest_common_prefix.py
# you can write to stdout for debugging purposes, e.g.
# print("this is a debug message")

def solution(A):
    # write your code in Python

    max_substring_len = len(A[0])
    for i in range(1,len(A)):
        if len(A[i]) < max_substring_len:
            max_substring_len = len(A[i])


    for i in range(1,len(A)):
        for j in range(max_substring_len):
            if A[0][j] != A[i][j]:
                max_substring_len -= 1
                if (max_substring_len == 0) : return ""

    return A[0][0:max_substring_len]
